planner/filter/complex.go

```diff
@@ -32,6 +32,7 @@ func isComplex(conditions any, seekRelation bool) bool {
 		for k, v := range typedCond {
 			if op, ok := k.(*mapper.Operator); ok {
 				if (op.Operation == request.FilterOpOr && len(v.([]any)) > 1) ||
```

Comment:
suggestion: I think this can be simplified a bit and avoid the potential panic from type casting.

```go
isOpComplex := false
switch op.Operation {
  case request.FilterOpOr:
    isOpComplex = isComplex(v, true)
  case request.FilterOpAnd:
    isOpComplex = isComplex(v, true)
  case request.FilterOpNot:
    isOpComplex = isComplex(v, true)
}
if isOpComplex {
  return true
}
continue
```

Reply:
`v` can't be anything other than an `[]any` if it's an `_or` or `_and` operator and we don't want to classify them as complex if there is only one element in the array.

Reply:
I don't think this changes the logic though. It just adds another recursive step.

Reply:
made a change in line with what you suggested. Let me know what you think.